### PR TITLE
generate gems.rb in bundle init with feature flag

### PR DESCRIPTION
**Pull request by colby-swandale**
This PR adds a feature flag to generate a new `gems.rb` with `bundle init`.

### CODE REVIEW

File: lib/bundler/cli/init.rb [hunk 1]
```diff
@@ -21,15 +21,21 @@ def run
 
         spec = Bundler.load_gemspec_uncached(gemspec)
 
-        puts "Writing new Gemfile to #{SharedHelpers.pwd}/Gemfile"
-        File.open("Gemfile", "wb") do |file|
+        puts "Writing new Gemfile to #{SharedHelpers.pwd}/#{gemfile}"
```
Comment (segiddins) [inline lib/bundler/cli/init.rb]: Probably should replace Gemfile in the string as well

File: lib/bundler/feature_flag.rb [hunk 2]
```diff
@@ -18,6 +18,7 @@ def self.settings_flag(flag, &default)
     settings_flag(:only_update_to_newer_versions) { bundler_2_mode? }
     settings_flag(:plugins) { @bundler_version >= Gem::Version.new("1.14") }
     settings_flag(:error_on_stderr) { bundler_2_mode? }
+    settings_flag(:new_gemfile_name) { bundler_2_mode? }
```
Comment (segiddins) [inline lib/bundler/feature_flag.rb]: I feel this name should have `gems.rb` somewhere in it

**Comment by segiddins** [comment]
Looks good! After this, we should make an issue to have the same feature flag control the order in which we prefer one to the other when automatically discovering a gemfile.

File: lib/bundler/settings.rb [hunk 3]
```diff
@@ -28,6 +28,7 @@ class Settings
       plugins
       silence_root_warning
       error_on_stderr
+      init_gems_rb
```
Comment (segiddins) [inline lib/bundler/settings.rb]: can we put this in alphabetical order, please?

Comment (colby-swandale) [inline lib/bundler/settings.rb]: ah oops, certainly.

**Comment by colby-swandale** [comment]
ping @segiddins

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**